integration/kubernetes/vfio.sh

```diff
@@ -0,0 +1,153 @@
+		cp -a "${default_config_file}" "${SYSCONFIG_FILE}"
+	else
+		die "Error: configuration file for ${hypervisor} doesn't exist"
+	fi
+
+	# machine type applies to configuration.toml and configuration-qemu.toml
+	if [ -n "${machine_type}" ]; then
+		if [ "${hypervisor}" = "qemu" ]; then
+			sed -i 's|^machine_type.*|machine_type = "'${machine_type}'"|g' "${SYSCONFIG_FILE}"
+		else
+			info "Variable machine_type only applies to qemu. It will be ignored"
+		fi
+	fi
+
+	if [ -n "${sandbox_cgroup_only}" ]; then
+		sed -i 's|^sandbox_cgroup_only.*|sandbox_cgroup_only='${sandbox_cgroup_only}'|g' "${SYSCONFIG_FILE}"
+	fi
+
+	# Change to initrd or image depending on user input.
+	# Non-default configs must be changed to specify either initrd or image, image is default.
+	if [ "${image_type}" = "initrd" ]; then
+		if $(grep -q "^image.*" "${SYSCONFIG_FILE}"); then
+			if $(grep -q "^initrd.*" "${SYSCONFIG_FILE}"); then
+				sed -i '/^image.*/d' "${SYSCONFIG_FILE}"
+			else
+				sed -i 's|^image.*|initrd = "'${initrd_file}'"|g' "${SYSCONFIG_FILE}"
+			fi
+		fi
+	else
+		if $(grep -q "^initrd.*" "${SYSCONFIG_FILE}"); then
+			if $(grep -q "^image.*" "${SYSCONFIG_FILE}"); then
+				sed -i '/^initrd.*/d' "${SYSCONFIG_FILE}"
+			else
+				sed -i 's|^initrd.*|image = "'${image_file}'"|g' "${SYSCONFIG_FILE}"
+			fi
+		fi
+	fi
+
+	# enable debug
+	sed -i -e 's/^#\(enable_debug\).*=.*$/\1 = true/g' \
+		-e 's/^kernel_params = "\(.*\)"/kernel_params = "\1 agent.log=debug"/g' \
+		"${SYSCONFIG_FILE}"
+}
+
+run_test() {
+	setup_configuration_file "$1" "$2" "$3" "$4"
+
+	sudo -E kubectl apply -f "${SCRIPT_DIR}/runtimeclass_workloads/vfio.yaml"
+
+	pod_name=vfio
+	sudo -E kubectl wait --for=condition=Ready pod "${pod_name}"
+
+	net_interfaces=$(sudo -E kubectl exec -ti "${pod_name}" ip a | egrep -o "eth[0-9]+:" | wc -l)
+	if [ ${net_interfaces} -ne 2 ]; then
+		die "Error: expecting 2 network interfaces (eth0 and eth1), Got: $(kubectl exec -ti "${pod_name}" ip a)"
+	else
+		info "Success: found 2 network interfaces"
+	fi
+}
+
+main() {
+	# Init k8s cluster
+	${SCRIPT_DIR}/init.sh
+
+	# Create VFIO device
+	ADDR="00:03.0"
+	sudo modprobe vfio
+	sudo modprobe vfio-pci
+	echo 0000:${ADDR} | sudo tee /sys/bus/pci/devices/0000:${ADDR}/driver/unbind
+	echo '1af4 1041' | sudo tee /sys/bus/pci/drivers/vfio-pci/new_id
+
+	# Install network (device) plugin
+	git clone --depth=1 https://github.com/intel/sriov-network-device-plugin
+	pushd sriov-network-device-plugin
+	sed -i 's|resourceList.*|resourceList": [{"resourceName":"virtio_net","selectors":{"vendors":["1af4"],"devices":["1041"],"drivers":["vfio-pci"],"pfNames":["eth1"]}},{|g' deployments/configMap.yaml
+	sudo -E kubectl create -f deployments/configMap.yaml
+	sudo -E kubectl create -f deployments/k8s-v1.16/sriovdp-daemonset.yaml
+	sleep 5
+	popd
+	rm -rf sriov-network-device-plugin
+
+	sriov_pod="$(sudo -E kubectl --namespace=kube-system get pods --output=name | grep sriov-device-plugin | cut -d/ -f2)"
```

Comment:
`grep sriov-device-plugin`: is this correct, or should it be `sriov-network-device-plugin`?

Reply:
yes, it is